pasqal_cloud/__init__.py

```diff
@@ -230,6 +231,81 @@ def get_batch(self, id: str, fetch_results: bool = False) -> Batch:
         self.batches[batch.id] = batch
         return batch
 
+    def get_batches(
+        self,
+        filters: Optional[BatchFilters] = None,
+        pagination_params: Optional[PaginationParams] = None,
+    ) -> PaginatedResponse:
+        """
+        Retrieve a list of batches matching filters using a pagination system.
+
+        batches are sorted by their creation date in descending order.
+
+        If no 'pagination_params' are provided, the first 100 batches
+        matching the query will be returned by default.
+
+        Args:
+            filters: Filters to be applied to the batches.
+            pagination_params: Pagination to be applied to the query.
+
+        Examples:
+        >>> get_batches(filters=BatchFilters(status=BatchStatus.ERROR))
+        Returns the first 100 batches with an ERROR status.
+
+        >>> get_batches(filters=BatchFilters(status=BatchStatus.ERROR),
+                     pagination_params=PaginationParams(offset=100))
+        Returns batches 101-200 with an ERROR status.
+
+        >>> get_batches(filters=BatchFilters(status=BatchStatus.ERROR),
+                     pagination_params=PaginationParams(offset=200))
+        Returns batches 201-300 with an ERROR status.
+
+        Returns:
+            PaginatedResponse: A class instance with two fields:
+                - total: An integer representing the total number of batches
+                         matching the filters.
+                - results: A list of batches matching the filters and pagination
+                           parameters provided.
+
+        Raises:
+            BatchFetchingError: If fetching batches call failed.
+            ValueError: If `filters` is not an instance of BatchFilters
+                    or if `pagination_params` is not an instance of PaginationParams.
+        """
+
+        if pagination_params is None:
+            pagination_params = PaginationParams()
+        elif not isinstance(pagination_params, PaginationParams):
+            raise ValueError(
```

Comment:
nit: `TypeError` seems more appropriate

Reply:
You are right, I used Valuerror on my previous PRs related to bulk actions, should I update them you think ?